### PR TITLE
[App Service] `az webapp`: Add Java 17 Support

### PR DESCRIPTION
**Pull request by StrawnSC**
**Related command**

`az webapp list-runtimes`, `az webapp create`, `az webapp up` 

**Description**

Adds support for creating java 17 webapps 

**Testing Guide**

`az webapp list-runtimes`

**History Notes**

[App Service] `az webapp list-runtimes`: Add Java 17 Support
[App Service] `az webapp create`: Add Java 17 Support
[App Service] `az webapp up`: Add Java 17 Support

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by panchagnula** [comment]
@StrawnSC what about functionapp? I am expecting the changes you made are common across both?

**Comment by StrawnSC** [comment]
> @StrawnSC what about functionapp? I am expecting the changes you made are common across both?

I checked functions and there isn't any change needed there. They use pretty different APIs (there's no special java handling in function apps) and there isn't java 17 support in function apps

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -2896,7 +2896,7 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
 class _AbstractStackRuntimeHelper:
     def __init__(self, cmd, linux=False, windows=False):
         self._cmd = cmd
-        self._client = web_client_factory(cmd.cli_ctx, api_version="2021-01-01")
+        self._client = web_client_factory(cmd.cli_ctx, api_version="2021-01-15")
```
[the remark this answers is not in the document]
Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:2899]: I just tried this with the latest/default API version and the API returns a 500 error. The error message doesn't seem helpful (just says "an error has occurred"). I'll loop Aniello in on this

[the remark this answers is not in the document]
Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:2899]: oh wow! I somehow thought this issue was resolved. Not a blocker for this change btw!

**Review by panchagnula — APPROVED** [review]
Code changes look good to me. Please get the build working (like you already are)  & w e should good to merge. Thank you.

**Comment by yonzhan** [comment]
App Service

**Comment by StrawnSC** [comment]
@zhoxing-ms is there any chance we could get this in the upcoming release? It's a pretty small change and should be good to go

I'm not sure why the Azure.azure-cli CI check is failing. I've fixed all of the failing tests

**Review by zhoxing-ms — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**